src/main/java/com/lineate/testyourlexicon/services/GameService.java

```diff
@@ -7,17 +7,25 @@
 import com.lineate.testyourlexicon.repositories.UserRepository;
 import com.lineate.testyourlexicon.util.GameMapper;
 import java.util.Arrays;
+import java.util.List;
+
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class GameService {
   private final UserRepository userRepository;
+  private final LanguageService languageService;
 
   public GameConfigurationDto configure(GameConfigurationDto gameConfigurationDto, User user) {
     GameConfiguration gameConfiguration =
         GameMapper.gameConfigurationDtoToGameConfiguration(gameConfigurationDto);
+    List<String> supportedLanguages = languageService.supportedLanguages();
+    if (!supportedLanguages.contains(gameConfigurationDto.getTranslateFrom()))
+      throw new IllegalArgumentException("unsupported language");
+    if (!supportedLanguages.contains(gameConfigurationDto.getTranslateTo()))
+      throw new IllegalArgumentException("unsupported language");
```

Comment:
Let's combine these into a single condition

```suggestion
    if (!supportedLanguages.contains(gameConfigurationDto.getTranslateFrom()) || 
        !supportedLanguages.contains(gameConfigurationDto.getTranslateTo())) {
        throw new IllegalArgumentException("unsupported language");
    }
```